### PR TITLE
Fix/useStore

### PR DESCRIPTION
**Pull request by zarabotaet**
Two different workarounds in two commits

### CODE REVIEW

**Comment by zarabotaet** [comment]
Because useState is a wrapper  around useReducer we can directly take useReducer.
https://codesandbox.io/s/hopeful-lovelace-qeufx?file=/src/App.js
Sandbox to play with the proposed solution.

File: src/react/__tests__/useStore.test.tsx [hunk 1]
```diff
@@ -90,6 +90,29 @@ describe('useStore', () => {
...
     `)
   })
+  it('should correct work, when store contains function', async () => {
+    const fn = jest.fn()
+    const changeStore = createEvent()
+    const $store = createStore(() => 0).on(changeStore, (_, p) => p)
+
+    const Display = () => {
+      const store = useStore($store)
+      fn()
```
Comment (zerobias) [inline src/react/__tests__/useStore.test.tsx:100]: There should be a proof that issues does exists at all: your example [reveal](https://codesandbox.io/s/admiring-goldberg-i9q5l?file=/src/App.js) that current implementation works as expected

**Review by zerobias — CHANGES_REQUESTED** [review]

File: src/react/__tests__/useStore.test.tsx (hunk 1 above)
Comment (zarabotaet) [reply src/react/__tests__/useStore.test.tsx:100]: [original issue](https://share.effector.dev/5T7AGosk) 
[new sandbox with proofs](https://codesandbox.io/s/sleepy-shadow-b9nf4?file=/src/App.js)

**Review by zerobias — APPROVED** [review]
All good, big thanks 👍 👍

**Pull request merged**